server/controllers/finance/trialBalance/index.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * The trial balance provides a description of what the general
+ * ledger would look like after posting data from the
+ * posting journal to the general ledger.
+ * It also submit errors back to the client.
+ *
+ */
+
+var q = require('q'),
+    db = require('../../../lib/db'),
+    uuid = require('node-uuid'),
+    util = require('../../../lib/util'),
+    NotFound   = require('../../../lib/errors/NotFound'),
+    BadRequest = require('../../../lib/errors/BadRequest'),
+    _ = require('lodash');
+
+// utility function to sum an array of objects on a particular property
+function aggregate(property, array) {
+  return array.reduce(function (s, row) {
+    return s + row[property];
+  }, 0);
+}
+
+// creates an error report for a given code
+function createErrorReport(code, isFatal, rows) {
+  return {
+    code : code,
+    fatal : isFatal,
+    transactions : rows.map(function (row) { return row.trans_id; }),
+    affectedRows : aggregate('count', rows)
+  };
+}
+
+// Warning if the entity is null and entity_type is null also
+function checkEntityIsAlwaysDefined(transactions) {
+
+  var sql =
+    `SELECT COUNT(pj.uuid) AS count, pj.trans_id, pj.entity_uuid, pj.entity_type FROM posting_journal AS pj
+    WHERE pj.trans_id IN (?) AND pj.entity_type IS NULL 
+    GROUP BY trans_id HAVING pj.entity_uuid IS NULL;`;
+
+  return db.exec(sql, [transactions])
+    .then(function (rows) {
+
+      // if nothing is returned, skip error report
+      if (!rows.length) { return; }
+
+      // returns a error report
+      return createErrorReport('POSTING_JOURNAL.WARNINGS.MISSING_ENTITY', false, rows);
+    });
+}
+
+// make sure that a entity_uuid exists for each deb_cred_type
+function checkDescriptionExists(transactions) {
+
+  var sql =
+    `SELECT COUNT(pj.uuid) AS count, pj.trans_id, pj.description FROM posting_journal AS pj
+    WHERE pj.trans_id IN (?) GROUP BY trans_id HAVING pj.description IS NULL;`;
+
+  return db.exec(sql, [transactions])
+  .then(function (rows) {
+
+    // if nothing is returned, skip error report
+    if (!rows.length) { return; }
+
+    // returns a error report
+    return createErrorReport('POSTING_JOURNAL.ERRORS.MISSING_DESCRIPTION', true, rows);
+  });
+}
+
+// make sure that a entity_uuid exists for each deb_cred_type
+function checkEntityExists(transactions) {
+
+  var sql =
+    `SELECT COUNT(pj.uuid) AS count, pj.trans_id, pj.entity_uuid FROM posting_journal AS pj
+    WHERE pj.trans_id IN (?) AND (pj.entity_type = 'D' OR pj.entity_type = 'C')
+    GROUP BY trans_id HAVING pj.entity_uuid IS NULL;`;
+
+  return db.exec(sql, [transactions])
+    .then(function (rows) {
+
+      // if nothing is returned, skip error report
+      if (!rows.length) { return; }
+
+      // returns a error report
+      return createErrorReport('POSTING_JOURNAL.ERRORS.MISSING_ENTITY', true, rows);
+    });
+}
+
+// make sure that the document Id exist in each line of the transaction
+function checkDocumentIDExists(transactions) {
```

Comment:
The column `document_id` doesn't exist anymore.  So a better name is probably `checkRecordUuidExists()`.  The comment could also be updated to reflect the removal of `document_id`.